docs/source/routing/observability/debugging-client-requests.mdx

```diff
@@ -0,0 +1,153 @@
+---
+title: Debugging Client Requests to GraphOS Router
+subtitle: Options for analyzing and debugging incoming requests
+description: Learn how to use GraphOS router telemetry and GraphOS Insights to inspect and debug incoming HTTP client requests.
+context:
+  - telemetry
+---
+
+By default, the GraphOS Router operates [without generating HTTP request logs or exporting telemetry metrics beyond what it sends to GraphOS](/graphos/routing/observability).
+This default minimizes potentially high observability costs that can result from high request volumes.
+If you need more data than the default [GraphOS Insights](/graphos/platform/insights), you can configure your router to collect and export additional telemetry.
+
+## Using GraphOS Insights
+
+GraphOS Studio lets you analyze data from failed requests, such as GraphQL error messages ([if enabled](/graphos/routing/graphos-reporting#errors)) and the ID of the client making the request. You can also [segment your insights data](/graphos/platform/insights/client-segmentation) based on the client ID.
+
+<Tip>
+
+[Learn how to ensure client IDs are included in all requests.](/graphos/routing/observability/client-id-enforcement)
+
+</Tip>
+
+## Enabling additional telemetry
+
+Use the methods below if you need information outside of what's presented in GraphOS Studio to debug client requests.
+
+<Note>
+
+If you want to debug client requests in your own environment, Apollo recommends first doing so in a non-production environment or using logic to debug on a per-request basis.
+
+</Note>
+
+### Telemetry traces and spans
+
+Once you enable [router telemetry](/graphos/routing/observability/telemetry), traces can have spans labeled `router`. You can add attributes to include helpful information about the request and response. The following sections show some common patterns.
+
+<Note>
+
+Spans don't include an attribute for the HTTP body text. If you want to see the GraphQL operation, first check your [GraphOS Insights](/graphos/platform/insights) or use the [request logging configuration](#request-body-logging).
+</Note>
+
+#### Add OpenTelemetry attributes
+
+Add [OpenTelemetry (OTel) standard attributes](/graphos/reference/router/telemetry/instrumentation/standard-attributes#router) such as `error.type` and `http.response.status_code` to spans with this configuration:
+
+```yaml title="router.yaml" showLineNumbers=false
+telemetry:
+  instrumentation:
+    spans:
+      router:
+        attributes:
+          http.response.status_code: true
+          error.type: true
+```
+
+#### Map GraphQL errors to OTel errors
+
+Map GraphQL error data to OTel errors with the following configuration:
+
+```yaml title="router.yaml" showLineNumbers=false
+telemetry:
```

Comment:
Hmm, it seems [the commit](https://github.com/apollographql/router/pull/6563/commits/137c4d157fff22bcc67c3db0e5978a5f3de83058) that adds these configs is causing Circle CI tests that validate configs to [fail](https://app.circleci.com/pipelines/github/apollographql/router/30388/workflows/42aea861-0b86-4696-a1e7-104a3f87b984/jobs/223452). 

<img width="819" alt="image" src="https://github.com/user-attachments/assets/5c47f978-2be1-405b-add2-b189bfd8e1ea" />

When testing them in VS Code with [router config awareness](https://www.apollographql.com/docs/graphos/reference/router/configuration#configuration-awareness-in-your-text-editor), they didn't throw any issues though. 🤔